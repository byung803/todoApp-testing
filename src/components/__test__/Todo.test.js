import React from 'react';
import { shallow } from 'enzyme';
import Todo from '../Todo';

let wrapper, toggleTodoSpy, removeTodoSpy;

beforeEach(() => {
    toggleTodoSpy = jest.fn();
    removeTodoSpy = jest.fn();
    wrapper = shallow(<Todo toggleTodo={toggleTodoSpy} removeTodo={removeTodoSpy} />);
});

describe('Todo Compo', () => {
    it('should have a li and a button', () => {
        expect(wrapper.find('li').length).toBe(1);
        expect(wrapper.find('button').length).toBe(1);
    })

    it('should use toggleTodo when li is clicked', () => {
        wrapper.find('li').simulate('click');
        expect(toggleTodoSpy).toHaveBeenCalled();
    })

    it('should use removeTodo when button is clicked', () => {
        wrapper.find('button').simulate('click');
        expect(removeTodoSpy).toHaveBeenCalled();
    })
})