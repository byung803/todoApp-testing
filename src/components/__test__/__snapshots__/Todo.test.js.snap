// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo Compo should have snapshot 1`] = `
<div>
  <li
    onClick={[MockFunction]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
  />
  <button
    onClick={[MockFunction]}
  >
    remove
  </button>
</div>
`;
